django/contrib/postgres/aggregates/statistics.py

```diff
@@ -0,0 +1,120 @@
+from django.db.models import F, FloatField, IntegerField
+from django.db.models.aggregates import Aggregate
+from django.db.models.expressions import Value
+from django.utils import six
+
+__all_ = [
+    'CovarPop', 'Corr', 'RegrAvgX', 'RegrAvgY', 'RegrCount', 'RegrIntercept',
+    'RegrR2', 'RegrSlope', 'RegrSXX', 'RegrSXY', 'RegrSYY',
+]
+
+
+class StatFunc(Aggregate):
+    template = "%(function)s(%(y)s, %(x)s)"
+    _num_expression_alias = 'num'
+
+    def __init__(self, y, x, output_field=FloatField()):
+        if not x or not y:
+            raise TypeError('Both X and Y must be provided.')
+        if (not isinstance(x, (six.text_type, six.string_types)) or
+                not isinstance(y, (six.text_type, six.string_types))):
+            raise ValueError('X and Y must be a string.')
+        super(StatFunc, self).__init__(output_field=output_field)
+        self.x = x
+        self.y = y
+        self.source_expressions = self.parse_expressions([self.x, self.y])
+
+    def parse_expressions(self, expressions):
+        # Some stat functions allows integer to be an argument,
+        # so we need to parse it and not resolve expression as F for
+        # this case.
+        return [
+            Value(arg) if arg.isdigit() else F(arg) for arg in expressions
+        ]
```

Comment:
@coldmind I think @charettes was suggesting change the original `_parse_expressions` to differentiate between strings and numbers, and I think that's a great idea.

Rather than define the method above on `StatFunc`, we should change the method directly on `Func` which will then propagate to StatFunc through inheritance.

It may be a little magical, but the rules can be defined quite clearly. "string is a field ref", "resolve_expression passed verbatim", everything else is a "Value(value)".

I like this enough to write a patch today for inclusion in 1.8beta - and update all the applicable tests. Will land it today hopefully, and then we can rebase this PR here.